src/audio/mux/mux.c

```diff
@@ -383,427 +349,186 @@ static void set_mux_params(struct comp_dev *dev,
+		 pipe_id);
 	return 0;
 }
 
-/* set component audio stream parameters */
-static int mux_params(struct comp_dev *dev,
-		      struct sof_ipc_stream_params *params)
-{
-	int ret;
-
-	comp_info(dev, "mux_params()");
-#if CONFIG_IPC_MAJOR_4
-	set_mux_params(dev, params);
-#endif
-	ret = comp_verify_params(dev, BUFF_PARAMS_CHANNELS, params);
-	if (ret < 0)
-		comp_err(dev, "mux_params(): comp_verify_params() failed.");
-
-	return ret;
-}
-
-static int mux_ctrl_set_cmd(struct comp_dev *dev,
-			    struct sof_ipc_ctrl_data *cdata)
-{
-	struct comp_data *cd = comp_get_drvdata(dev);
-	struct sof_mux_config *cfg;
-	int ret = 0;
-
-	comp_info(dev, "mux_ctrl_set_cmd(), cdata->cmd = 0x%08x",
-		  cdata->cmd);
-
-	switch (cdata->cmd) {
-	case SOF_CTRL_CMD_BINARY:
-		cfg = (struct sof_mux_config *)
-		      ASSUME_ALIGNED(&cdata->data->data, 4);
-
-		ret = mux_set_values(dev, cd, cfg);
-		break;
-	default:
-		comp_err(dev, "mux_ctrl_set_cmd(): invalid cdata->cmd = 0x%08x",
-			 cdata->cmd);
-		ret = -EINVAL;
-		break;
-	}
-
-	return ret;
-}
-
-static int mux_ctrl_get_cmd(struct comp_dev *dev,
-			    struct sof_ipc_ctrl_data *cdata, int size)
-{
-	struct comp_data *cd = comp_get_drvdata(dev);
-	struct sof_mux_config *cfg = &cd->config;
-	uint32_t reply_size;
-	int ret = 0;
-
-	comp_cl_info(&comp_mux, "mux_ctrl_get_cmd(), cdata->cmd = 0x%08x",
-		     cdata->cmd);
-
-	switch (cdata->cmd) {
-	case SOF_CTRL_CMD_BINARY:
-		/* calculate config size */
-		reply_size = sizeof(struct sof_mux_config) + cfg->num_streams *
-			sizeof(struct mux_stream_data);
-
-		/* copy back to user space */
-		const int mux_memcpy_err __unused =
-			memcpy_s(cdata->data->data, ((struct sof_abi_hdr *)
-						     (cdata->data))->size, cfg, reply_size);
-		assert(!mux_memcpy_err);
-
-		cdata->data->abi = SOF_ABI_VERSION;
-		cdata->data->size = reply_size;
-		break;
-	default:
-		comp_cl_err(&comp_mux, "mux_ctrl_set_cmd(): invalid cdata->cmd = 0x%08x",
-			    cdata->cmd);
-		ret = -EINVAL;
-		break;
-	}
-
-	return ret;
-}
-
-/* used to pass standard and bespoke commands (with data) to component */
-static int mux_cmd(struct comp_dev *dev, int cmd, void *data,
-		   int max_data_size)
-{
-	struct sof_ipc_ctrl_data *cdata = ASSUME_ALIGNED(data, 4);
-
-	comp_info(dev, "mux_cmd() cmd = 0x%08x", cmd);
-
-	switch (cmd) {
-	case COMP_CMD_SET_DATA:
-		return mux_ctrl_set_cmd(dev, cdata);
-	case COMP_CMD_GET_DATA:
-		return mux_ctrl_get_cmd(dev, cdata, max_data_size);
-	default:
-		return -EINVAL;
-	}
-}
-
-static void mux_prepare_active_look_up(struct comp_dev *dev,
+static void mux_prepare_active_look_up(struct comp_data *cd,
 				       struct audio_stream __sparse_cache *sink,
 				       const struct audio_stream __sparse_cache **sources)
 {
-	struct comp_data *cd = comp_get_drvdata(dev);
 	const struct audio_stream __sparse_cache *source;
-	uint8_t active_elem;
-	uint8_t elem;
-
-	cd->active_lookup.num_elems = 0;
-
-	active_elem = 0;
+	int elem;
```

Comment:
This and rest of of changes in this function should go into a code simplification commit, not related to module adapter convert.